reflection.md

```diff
@@ -2,17 +2,51 @@
+   
+   
+ 
    + What doesn't work about this design? (For example, what things seem
    cumbersome to say?)
+   
+What seems a bit more cumbersome to say the following: " val telNumber = '(' ~ digit{3} ~ ')' ~ digit{3} ~ '-' ~ digit{4} ". However, given the flexibility and constraints we want users to be able to follow in creating a valid phone number, I honestly cannot think of a simpler solution that does not compromise either the flexibility and/or the constraints. The flexibility includes being able to choose any 3 digits between the parentheses, to choose any of the 3 digits before the dash, and to choose any 4 digits after the dash. The constraints include making sure that only (, ), -, and digits are included. Given the specific function of each element in the series of characters that compose a valid tedNumber, I can't think of any simpler way to say this (although this isn't a critique of the design -- this is my way to express how the design is "cumbersome" but as simple as I think it can get). Overall, the < > surrounding * and surrounding + make the language look cumbersome, but then again, I really can't come up with an alternative way to express what <*> and <+> express, and we definitely can't just remove the < > because just having * totally changes the meaning of <*>
+
+
+   
+ 
    + Think of a syntactic change that might make the language better. How would
    you implement it _or_ what features of Scala would prevent you from
    implementing it? (You don't have to write code for this part. You could say
    "I would use literal extension to..." or "Scala's rules for valid
-   identifiers prevent...")
\ No newline at end of file
+   identifiers prevent...")
+
+I actually have a question about how Scala itself is designed. Consider the two declarations:
+
+ def || (other: RegularExpression) = Union(this, other)
+
+and
+
+case class Literal(val literal: Char) extends RegularExpression .
+
+I think it's interesting that the argument for def || is "other: RegularExpression" whereas the argument for case class Literal is "val literal: Char". I knew to add "val" for case class Literal and not for def || because of look at previous examples, but I don't think I understand Scala enough to reason out, for myself, why one requires "val" and why the other does not. 
```

Comment:
The former is just defining a name to refer to something that will be passed into the method as a parameter, and the latter defines an immutable field in the case class. I'm not entirely sure, but you may be able to omit the `val` in the `Literal` definition since it is defined as a case class. (It compiles without it, but it may mean the underlying object is different.) @bwiedermann: Is the `val` needed when defining a case class?